src/components/Note.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { useDispatch } from 'react-redux'
+
+// components
+import ContentEditable from 'react-contenteditable'
+
+// action-creators
+import deleteAttribute from '../action-creators/deleteAttribute'
+import setAttribute from '../action-creators/setAttribute'
+
+// util
+import {
+  attribute,
+  isContextViewActive,
+  selectNextEditable,
+  setSelection,
+} from '../util.js'
+
+// gets the editable node for the given note element
+const editableOfNote = noteEl =>
+  noteEl.parentNode.previousSibling.querySelector('.editable')
+
+export const Note = ({ context }) => {
+
+  const note = attribute(context, '=note')
+
+  if (note === undefined || isContextViewActive(context)) return null
+
+  const dispatch = useDispatch()
+
+  return <div className='note children-subheading text-note text-small' style={{ top: '4px' }}>
+    <ContentEditable
+      html={note || ''}
+      placeholder='Enter a note'
+      onKeyDown={e => {
+
+        // delete empty note
+        // need to get updated note attribute (not the note in the outside scope)
+        const note = attribute(context, '=note')
+
+        // select thought
+        if (e.key === 'Escape' || e.key === 'ArrowUp' || (e.metaKey && e.altKey && e.keyCode === 'N'.charCodeAt(0))) {
```

Comment:
Hello @raineorshine ,
I think this is too much logic to be put on a return statement. I think it would be better if you had a function that handles these key events and then just pass it as a prop to onkeyDown. What do you think?

Reply:
I am neutral on this one. As you may see throughout the codebase, I am generally happy to have lots of code inline, as it reduces the number of variables to reason about and helps the structure of the code better reflect the control flow. It often requires breaking it up into multiple lines at least.